core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -338,7 +338,7 @@ private[spark] class KryoSerializerInstance(ks: KryoSerializer, useUnsafe: Boole
 
   // Make these lazy vals to avoid creating a buffer unless we use them.
   private lazy val output = ks.newKryoOutput()
-  private lazy val input = if (useUnsafe) new KryoUnsafeInput() else new KryoInput()
+  private lazy val input = if (useUnsafe) new KryoUnsafeInput(4096) else new KryoInput(4096)
```

Comment:
why do we need to set the buffer size here while that's set by `setBuffer`? Can't we set it only when the `ByteBuffer` is backed by an accessible byte array?

Reply:
Yeah, this will be a wasted allocation in the 'hasArray' case, and should be set below.